routing/pathfind.go

```diff
@@ -665,26 +659,25 @@ func findPath(g *graphParams, r *RestrictParams, cfg *PathFindingConfig,
 		}
 	}
 
-	// If the source node isn't found in the next hop map, then a path
-	// doesn't exist, so we terminate in an error.
-	if _, ok := next[source]; !ok {
-		return nil, newErrf(ErrNoPathFound, "unable to find a path to "+
-			"destination")
-	}
-
-	// Use the nextHop map to unravel the forward path from source to
+	// Use the distance map to unravel the forward path from source to
 	// target.
 	pathEdges := make([]*channeldb.ChannelEdgePolicy, 0)
 	currentNode := source
 	for currentNode != target { // TODO(roasbeef): assumes no cycles
 		// Determine the next hop forward using the next map.
-		nextNode := next[currentNode]
+		currentNodeWithDist, ok := distance[currentNode]
+
+		if !ok || currentNodeWithDist.nextHop == nil {
```

Comment:
is the `currentNodeWithDist.nextHop == nil` case expected to ever happen? Might be worth returning a different error in that case.

Reply:
Well actually, no! It was expected before removing the `distance` prepopulation. What are you thinking in terms of different error? It's probably a programming error.

Reply:
If it is always sent when adding nodes to the distance map, we can just assume it is there I think